lib/ui/layout.js

```diff
@@ -1,101 +1,87 @@
-    dismissable: true
-  })
+const repl = () => {
+  return require('../runtime').console2
 }
-
-function getPanes() {
-  if (atom.workspace.getBottomDock) {
-    return atom.workspace.getPanes().slice(0, -3)
-  } else {
-    return atom.workspace.getPanes()
-  }
+const workspace = () => {
+  return require('../runtime').workspace
 }
-
-export function isCustomLayout() {
-  return getPanes().length > 1
+const plotPane = () => {
+  return require('../runtime').plots
 }
-
-function resetLayout() {
-  pane = getPanes()[0]
-  for (let p of getPanes().slice(1)) {
-    for (let item of p.getItems()) {
-      p.moveItemToPane(item, pane)
-    }
-  }
+const debuggerPane = () => {
+  return require('../runtime').debugger
+}
+const linter = () => {
+  return require('../runtime').linter
+}
+const documentation = () => {
+  return require('../ui').docpane
 }
 
-function createLayout(layout,
-                      pane = atom.workspace.getActivePane()) {
-  if (!layout){
-  } else if (layout.split && layout.items) {
-    let ps = [pane]
-    ps.push(layout.vertical ? pane.splitDown() : pane.splitRight())
-    for (let i = 0; i < ps.length; i++) {
-      createLayout(layout.items[i], ps[i])
-    }
-  } else if (!layout.split && layout.items) {
-    for (let i = 0; i < layout.items.length; i++) {
-      pane.activateItem(layout.items[i])
-    }
-  } else if (layout instanceof Array) {
-    for (let i = 0; i < layout.length; i++) {
-      pane.activateItem(layout[i])
-    }
-  } else {
-    pane.activateItem(layout)
-  }
+export function defaultLayout () {
+  const activeContainer = atom.workspace.getActivePaneContainer()
+  const activePane = atom.workspace.getActivePane()
+  const activeItem = atom.workspace.getActivePaneItem()
+
+  const panes = []
+  if (atom.config.get('julia-client.uiOptions.layouts.defaultPanes.console')) panes.push(repl)
+  if (atom.config.get('julia-client.uiOptions.layouts.defaultPanes.workspace')) panes.push(workspace)
+  if (atom.config.get('julia-client.uiOptions.layouts.defaultPanes.documentation')) panes.push(documentation)
+  if (atom.config.get('julia-client.uiOptions.layouts.defaultPanes.plotPane')) panes.push(plotPane)
+  if (atom.config.get('julia-client.uiOptions.layouts.defaultPanes.debuggerPane')) panes.push(debuggerPane)
+  if (atom.config.get('julia-client.uiOptions.layouts.defaultPanes.linter')) panes.push(linter)
+
+  openPanes(panes)
+
+  // Re-activate the initially activated item
+  activeContainer.activate()
+  activePane.activate()
+  activePane.activateItem(activeItem)
 }
 
-function workspace() { return require('../runtime').workspace.ws }
-function docs() { return require('../ui').docpane.pane }
-function cons() {
-  if (atom.config.get('julia-client.consoleOptions.consoleStyle') == 'REPL-based') {
-    return require('../runtime').console2.terminal
-  } else {
-    return require('../runtime').console.c
+function openPanes (panes) {
```

Comment:
What happens if an unexpected pane item is focussed when this runs?

Reply:
Umm could you tell me an idea about an example of "an unexpected paen item" ??

Reply:
![Peek 2019-06-07 20-55](https://user-images.githubusercontent.com/6735977/59126955-aa662080-8966-11e9-8f8f-de2ce7445f70.gif)

In that gif the focus initially is on the editor in the bottom right, which leads to a kinda funky layout. I think the previous implementation closed all panes first and then started opening the Juno panes.

Reply:
Daaaaamn, seriously funny state...
I thought not closing all the pane would be better in a sense that even with this command an user can keep a part of the previous state, but sure, closing all the panes seem better....